include/range/v3/numeric/partial_sum.hpp

```diff
@@ -50,19 +48,21 @@ namespace ranges
             tagged_pair<tag::in(I), tag::out(O)>
             operator()(I begin, S end, O result, S2 end_result, BOp bop = BOp{}, P proj = P{}) const
             {
-                using V = value_type_t<I>;
-                using X = concepts::Invocable::result_t<P&, V>;
-                coerce<V> v;
-                coerce<X> x;
-
+                using X = projected<projected<I, coerce<value_type_t<I>>>, P>;
+                using Y = reference_t<X>;
+                coerce<value_type_t<I>> v;
                 if(begin != end && result != end_result)
                 {
-                    auto t(x(invoke(proj, v(*begin))));
+                    auto &&tmp1 = *begin;
+                    auto &&tmp2 = v((decltype(tmp1) &&) tmp1);
+                    detail::decay_t<Y> t(invoke(proj, (decltype(tmp2) &&) tmp2));
```

Comment:
In theory, passing an actual value (as opposed to a proxy) to a projection should yield another honest-to-god value, as opposed to a proxy reference. Part of my reasoning for requiring `SemiRegular<decay_t<Y>>` is to get closer to enforcing that. `Regular`-ish types should have independent copies.